### DISCUSSION TITLE
Polling request taking on average 25 seconds to respond 200 status

### DISCUSSION THREAD

**Discussion by codertushar** [socketio/socket.io#4606] [Q&A] [answered]
can anybody give some idea, what can be the problem?
This is a Node-based service(only 1 instance) being used by around 200 customers.(stats about last 24 hours)

<img width="742" alt="image" src="https://user-images.githubusercontent.com/59651629/213979640-05fcd958-91fb-4271-a271-98cccb26fe0b.png">
(roughly 6000 requests  out of 53000 are taking around 25 seconds to fulfill)

To tell about the scale of request coming with this particular service in last 24 hours, here it is:
<img width="238" alt="image" src="https://user-images.githubusercontent.com/59651629/213980165-658ea5a1-ac69-44a3-9df6-ac86c04bbec3.png">

**Comment by darrachequesne** [comment, marked as answer]
Hi! That's how HTTP long-polling works: the server keeps the HTTP request open until it has some data to send to the client.

25 seconds is the default value for the heartbeat (which periodically checks the status of the connection).

References:

- https://socket.io/docs/v4/how-it-works/
- https://socket.io/docs/v4/server-options/#pinginterval

**Reply by codertushar** [reply]
Thanks for the explanation. 👍🏻

**Comment by codertushar** [comment]
@darrachequesne One more question, I have: why do some polling requests take a maximum of around 45 seconds or so?
<img width="724" alt="image" src="https://user-images.githubusercontent.com/59651629/214043830-2ed0fecc-5c2f-460c-8136-4c5d51a0d326.png">
If we have 25 seconds pingInterval, then why do some requests go until 45 seconds?

This is a chart for max_duration a request took to respond:
<img width="355" alt="image" src="https://user-images.githubusercontent.com/59651629/214044089-5b2a9272-e90d-41db-8379-61fe8d41a4e5.png">

Also, I noticed, some are taking hours even, what could be a possible theory?:
<img width="689" alt="image" src="https://user-images.githubusercontent.com/59651629/214047134-65c4f422-9abf-49c7-be84-7328065bbfbd.png">
could it be that connection remains open because of user switched his system off or his internet went off or the browser tab is closed?

**Reply by darrachequesne** [reply]
> some are taking hours even, what could be a possible theory?

That seems weird indeed. Maybe a big packet on a reaaaaally slow connection?

> could it be that connection remains open because of user switched his system off or his internet went off or the browser tab is closed?

I don't think so, the server should close the connection when the heartbeat fails.
